atomic_reactor/plugins/koji_import.py

```diff
@@ -303,41 +304,76 @@ def set_pnc_build_metadata(self, extra):
-                ]
-                extra["image"]["remote_sources"] = remote_sources_image_metadata
-
-                remote_sources_typeinfo_metadata = [
-                    {
-                        "name": remote_source["name"],
-                        "url": remote_source["url"].rstrip('/download'),
-                        "archives": [
-                            remote_source["remote_source_json"]["filename"],
-                            remote_source["remote_source_tarball"]["filename"],
-                            remote_source["remote_source_json_env"]["filename"],
-                            remote_source["remote_source_json_config"]["filename"],
-                        ],
-                    }
-                    for remote_source in remote_source_result
-                ]
-            else:
-                extra["image"]["remote_source_url"] = remote_source_result[0]["url"]
-                remote_sources_typeinfo_metadata = {
-                    "remote_source_url": remote_source_result[0]["url"]
-                }
+    def set_remote_sources_metadata_cachito(self, remote_source_result, extra):
+        if self.workflow.conf.allow_multiple_remote_sources:
+            remote_sources_image_metadata = [
+                {"name": remote_source["name"], "url": remote_source["url"].rstrip('/download')}
+                for remote_source in remote_source_result
+            ]
+            extra["image"]["remote_sources"] = remote_sources_image_metadata
 
-            remote_source_typeinfo = {
-                KOJI_BTYPE_REMOTE_SOURCES: remote_sources_typeinfo_metadata,
+            remote_sources_typeinfo_metadata = [
+                {
+                    "name": remote_source["name"],
+                    "url": remote_source["url"].rstrip('/download'),
+                    "archives": [
+                        remote_source["remote_source_json"]["filename"],
+                        remote_source["remote_source_tarball"]["filename"],
+                        remote_source["remote_source_json_env"]["filename"],
+                        remote_source["remote_source_json_config"]["filename"],
+                    ],
+                }
+                for remote_source in remote_source_result
+            ]
+        else:
+            extra["image"]["remote_source_url"] = remote_source_result[0]["url"]
+            remote_sources_typeinfo_metadata = {
+                "remote_source_url": remote_source_result[0]["url"]
             }
-            extra.setdefault("typeinfo", {}).update(remote_source_typeinfo)
+
+        remote_source_typeinfo = {
+            KOJI_BTYPE_REMOTE_SOURCES: remote_sources_typeinfo_metadata,
+        }
+        extra.setdefault("typeinfo", {}).update(remote_source_typeinfo)
+
+    def set_remote_sources_metadata_cachi2(self, remote_source_result, extra):
+        remote_sources_typeinfo_metadata = []
+        if self.workflow.conf.allow_multiple_remote_sources:
```

Comment:
It seems the `else` to this `if` is missing

Reply:
it seems but it's not, else is only to export Cachito URL, but we don't have that in cachi2. Dunno if I should export some other metadata in this case.

Even better would be to just allow multiple sources permanently, it would reduce lot of code

Reply:
Ah, right